src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -48,25 +49,70 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  private async checkWorkspacesExist(
+    finalOptions: SavedObjectsCreateOptions,
+    wrapperOptions: SavedObjectsClientWrapperOptions
+  ) {
+    if (finalOptions.workspaces?.length) {
+      let isAllTargetWorkspaceExisting = false;
+      // If only has one workspace, we should use get to optimize performance
+      if (finalOptions.workspaces.length === 1) {
+        const workspaceGet = await this.workspaceClient.get(
+          { request: wrapperOptions.request },
+          finalOptions.workspaces[0]
+        );
+        if (workspaceGet.success) {
+          isAllTargetWorkspaceExisting = true;
+        }
+      } else {
+        const workspaceList = await this.workspaceClient.list(
+          {
+            request: wrapperOptions.request,
+          },
+          {
+            perPage: 9999,
+          }
+        );
+        if (workspaceList.success) {
+          const workspaceIdsSet = new Set(
+            workspaceList.result.workspaces.map((workspace) => workspace.id)
+          );
+          isAllTargetWorkspaceExisting = finalOptions.workspaces.every((targetWorkspace) =>
+            workspaceIdsSet.has(targetWorkspace)
+          );
+        }
+      }
+
+      if (!isAllTargetWorkspaceExisting) {
+        throw SavedObjectsErrorHelpers.decorateBadRequestError(
+          new Error(
+            i18n.translate('workspace.id_consumer.invalid', {
```

Comment:
nit: do you think we could add an array like invalidWorkspaces to record it then error can output the invalid workspaces? do you think this is more helpful if we output like `Invalid workspaces: {workspaces}`?

Reply:
Thank you for the optimization you proposed. I will add the invalid workspaces list in the error output.

Reply:
Returning the workspace id maybe introduce XSS issue, so I will hidden the invalid workspace id.